### PR TITLE
Update supported version of sidecar

### PR DESCRIPTION
**Pull request by coulof**
Updated the versions of the side.

I used the following script to get the valid versions : https://gist.github.com/coulof/fe25743763a59e080e3f01a4717f5dc9

The logic happens in [end_of_life_grouped_versions](https://gist.github.com/coulof/fe25743763a59e080e3f01a4717f5dc9#file-get_supported_version_csi-sidecar-py-L42).

Fixes partially: #565

### CODE REVIEW

**Comment by xing-yang** [comment]
I changed "Fixes" to "Fixes partially" so that the issue won't get closed automatically.  We can use the same issue to track updates for other sidecars when they are ready.

File: book/src/external-provisioner.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
...
 [external-provisioner v3.4.1](https://github.com/kubernetes-csi/external-provisioner/releases/tag/v3.4.1) | [release-3.4](https://github.com/kubernetes-csi/external-provisioner/tree/release-3.4) | [v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0) | - | registry.k8s.io/sig-storage/csi-provisioner:v3.4.1 | v1.20 | - | v1.26
 [external-provisioner v3.3.1](https://github.com/kubernetes-csi/external-provisioner/releases/tag/v3.3.1) | [release-3.3](https://github.com/kubernetes-csi/external-provisioner/tree/release-3.3) | [v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0) | - | registry.k8s.io/sig-storage/csi-provisioner:v3.3.1 | v1.20 | - | v1.25
 [external-provisioner v3.2.1](https://github.com/kubernetes-csi/external-provisioner/releases/tag/v3.2.1) | [release-3.2](https://github.com/kubernetes-csi/external-provisioner/tree/release-3.2) | [v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0) | - | registry.k8s.io/sig-storage/csi-provisioner:v3.2.1 | v1.20 | - | v1.22
```
[the remark this answers is not in the document]
Comment (coulof) [reply book/src/external-provisioner.md]: good catch !

**Pull request merged**